### DISCUSSION TITLE
How to add a plugin via another plugin?

### DISCUSSION THREAD

**Discussion by hainessss** [TanStack/table#2535] [General]
Hello, 

I want to extend the useExpanded plugin by adding a column to the table that handles the expansion triggering.  I would like for this functionality to exist in one plugin so that the consumer only has to import one hook, so I am trying to build a custom plugin that adds the useExpanded and pushes a column to `visibleColumns`.  What is the best way to add a plugin to the table instance from another plugin?

Thanks in advance!

**Comment by gargroh** [comment]
One way of what i can think of you export array from ur plugin [ useExpanded, urCustomPlugin ] and consumer can import ur plugin and in useTable they can spread ...urPlugin.

**Comment by hainessss** [comment]
Any other suggestions, this seems like it should be a common use case?
